java/ql/src/Metrics/Summaries/GeneratedVsManualCoverage.ql

```diff
@@ -0,0 +1,64 @@
+/**
+ * @id java/summary/generated-vs-manual-coverage
+ * @name Metrics of generated versus manual MaD coverage
+ * @description Expose metrics for the number of API endpoints covered by generated versus manual MaD models.
+ * @kind table
+ * @tags summary
+ */
+
+import java
+import semmle.code.java.dataflow.FlowSummary
+import utils.modelgenerator.internal.CaptureModels
+
+/**
+ * Returns the number of `DataFlowTargetApi`s with Summary MaD models
+ * for a given package and provenance.
+ */
+bindingset[package]
+private float getNumMadModeledApis(string package, string provenance) {
+  provenance in ["generated", "manual", "both"] and
+  result =
+    count(SummarizedCallable sc |
+      package = sc.asCallable().getCompilationUnit().getPackage().getName() and
+      exists(DataFlowTargetApi dataFlowTargApi | sc.asCallable() = dataFlowTargApi) and
+      (
+        sc.isAutoGenerated() and // "auto-only"
+        provenance = "generated"
+        or
+        sc.isManuallyGenerated() and // "manual-only"
+        provenance = "manual"
+        or
+        sc.isBothAutoAndManuallyGenerated() and // "both"
+        provenance = "both"
+      )
+    )
+}
+
+/** Returns the total number of `DataFlowTargetApi`s for a given package. */
+bindingset[package]
+private float getNumApis(string package) {
+  result =
+    count(DataFlowTargetApi dataFlowTargApi |
+      package = dataFlowTargApi.getCompilationUnit().getPackage().getName()
+    )
+}
+
+from
+  DataFlowTargetApi dataFlowTargApi, string package, float generatedOnly, float both,
+  float manualOnly, float non, float all, float generatedCoverage, float manualCoverage
+where
+  // bind `package` to a `DataFlowTargetApi` package name
+  package = dataFlowTargApi.getCompilationUnit().getPackage().getName() and
```

Comment:
This line becomes superfluous with the above suggestions.

Reply:
I'm leaving this as is for now since I'm keeping `count` per yo-h's [comment](https://github.com/github/codeql/pull/11585#discussion_r1046384079). However, let me know if there is a better way to structure the binding for `package`.

Reply:
This binding is equivalent to the potential binding in `getNumApis`, so even if `getNumMadModeledApis` stays non-binding, then you can still make `getNumApis` binding and remove this line. As-is `getNumApis` will never yield 0 in combination with this line.